src/notifications/notificationmanager.cpp

```diff
@@ -512,8 +715,34 @@ QString NotificationManager::GetServerInformation(QString &vendor, QString &vers
 
 NotificationList NotificationManager::GetNotifications(const QString &owner)
 {
-    uint pid = callerProcessId();
-    QString callerProcessName = getProcessName(pid);
+    NotificationList notificationList;
+    if (!calledFromDBus()) {
+        notificationList = handleGetNotifications(getpid(), owner);
+    } else {
+        setDelayedReply(true);
+        ClientIdentifier *identifier = new ClientIdentifier(this, connection(), message());
+        connect(identifier, &ClientIdentifier::finished, this, &NotificationManager::identifiedGetNotifications, Qt::QueuedConnection);
+    }
+    return notificationList;
+}
+
+void NotificationManager::identifiedGetNotifications(ClientIdentifier *identifier)
+{
+    QVariantList arguments(identifier->message().arguments());
+    const QString owner = arguments.at(0).toString();
+    NotificationList notificationList = handleGetNotifications(identifier->clientPid(), owner);
+    if (identifier->message().isReplyRequired()) {
```

Comment:
Pondering should we care if someone calls a getter and declares no interest in the results. Doesn't feel like a sensical combination.

Reply:
No it is not sensical. But if somebody decides to do it anyway, sending a reply would be incorrect (and would lead to error logging by dbus daemon iirc) -> as we have taken responsibility for sending the reply, we have to check. But I would not care about it otherwise.

Reply:
Do we really need to check it though? I mean, I thought Qt is clever enough to not send it in the end. (It's always an option that I thought wrong.)

Reply:
Hm, had assumed this was more like reply can be omitted, but indeed the specification talks about "reply should be omitted". No matter, just a small detail so fine either way as long as we don't get warnings.

Reply:
Made a small test: skip the reply/not test, use dbus-send and dbus-monitor to observe what happens when reply is requested/not -> if we explicitly tell qt to connection.send(message), the message gets sent - which is sort of relief at least to me. Did not spot any warnings on journal - possibly verbosity settings have changed / differ between system and session bus? In any case IMO handling these in uniform manner is easier on brain (vs. spending time thinking case by case whether it is likely to be needed or not).